src/main/java/htsjdk/tribble/index/IndexFactory.java

```diff
@@ -306,8 +292,8 @@ public static LinearIndex createLinearIndex(final File inputFile, final FeatureC
             case INTERVAL_TREE: return createIntervalIndex(inputFile, codec);
             case LINEAR:        return createLinearIndex(inputFile, codec);
             case TABIX:         return createTabixIndex(inputFile, codec, sequenceDictionary);
+            default: throw new IllegalArgumentException("Unrecognized IndexType " + type);
```

Comment:
If this was a method call on `IndexType` then you wouldn't need a `switch` here or a `default` case.

Reply:
Yeah.  That's a really good point.  The whole structure here is super weird and could be rationalized.  I was doing the most braindead fix I could, but maybe I should move more stuff around.